src/stan/mcmc/hmc/hamiltonians/softabs_point.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef STAN_MCMC_HMC_HAMILTONIANS_SOFTABS_POINT_HPP
+#define STAN_MCMC_HMC_HAMILTONIANS_SOFTABS_POINT_HPP
+
+#include <stan/interface_callbacks/writer/base_writer.hpp>
+#include <stan/mcmc/hmc/hamiltonians/ps_point.hpp>
+
+namespace stan {
+  namespace mcmc {
+
+    // Point in a phase space with a base
+    // Riemannian manifold with SoftAbs metric
+    class softabs_point: public ps_point {
+    public:
+      explicit softabs_point(int n):
+        ps_point(n),
+        alpha(1.0),
+        hessian(Eigen::MatrixXd::Identity(n, n)),
+        eigen_deco(n),
+        log_det_metric(0),
+        softabs_lambda(Eigen::VectorXd::Zero(n)),
+        softabs_lambda_inv(Eigen::VectorXd::Zero(n)),
+        pseudo_j(Eigen::MatrixXd::Identity(n, n)) {}
+
+        // SoftAbs regularization parameter
+        double alpha;
```

Comment:
(fix now)  

1. The indenting is wrong for these member variables.

2.  The ones that are const (if any) should be marked as such.  It's as much for the reader of the code as the compiler and future writer of more code.